meson.build

```diff
@@ -0,0 +1,55 @@
+project('komorebi', 'vala', 'c',
+    version: '2.2.0',
+    meson_version: '>=0.49',
+)
+
+# Internationalisation / Localisation. This is currently only used to configure the .desktop files;
+i18n = import('i18n')
+po_dir = meson.source_root() / 'po'
+gettext_package = meson.project_name()
+add_project_arguments('-DGETTEXT_PACKAGE=' + gettext_package, language: 'c')
+i18n.gettext(meson.project_name(),
+    args: '--directory=' + meson.source_root()
+)
+
+# Set some variables
+prefix = get_option('prefix')
+bindir = get_option('bindir')
+datadir = get_option('datadir')
+pkgdatadir = join_paths(prefix, datadir, meson.project_name())
+
+# Configure Komorebi
+conf = configuration_data()
+conf.set_quoted('PACKAGE_NAME', meson.project_name())
+conf.set_quoted('PACKAGE_VERSION', meson.project_version())
+conf.set_quoted('PKGDATADIR', pkgdatadir)
+komorebiconfiguration = configure_file(input: 'config.vala.in', output: 'config.vala', configuration: conf)
```

Comment:
I would have done this by injecting the defines via the compiler but fair enough :grin:

Reply:
The Vala guys pointed me at this method for what we're doing, but if you can point me at some resources for doing it that way that I'd love to implement it more elegantly!

Reply:
No need. This is perfectly fine.